lms/djangoapps/courseware/views/views.py

```diff
@@ -1311,7 +1298,7 @@ def course_survey(request, course_id):
     """
 
     course_key = CourseKey.from_string(course_id)
-    course = get_course_with_access(request.user, 'load', course_key)
+    course = modulestore().get_course(course_key, depth=0)
```

Comment:
With moving the survey code into `has_acess`, using `get_course_with_access` on the `course_survey` view means that the learner can't access the survey for their course.